pkg/daemon/daemon.go

```diff
@@ -214,23 +215,40 @@ func (dn *Daemon) addProfileConfig(socketPath string, nodeProfile *ptpv1.PtpProf
 		nodeProfile.Ptp4lOpts = &opts
 	}
 
+	var output ptp4lconf;
 	if nodeProfile.Interface == nil || *nodeProfile.Interface == "" {
-		// We need to copy this to another variable because is a pointer
 		iface := string("")
 		nodeProfile.Interface = &iface
+	} else {
+		output.ifaceString = fmt.Sprintf("[%s]", *nodeProfile.Interface)
 	}
 
-	config := fmt.Sprintf("%s\nuds_address %s\nmessage_tag [%s]",
-		*nodeProfile.Ptp4lConf,
-		socketPath,
-		*nodeProfile.Interface)
-	nodeProfile.Ptp4lConf = &config
+	output = populatePtp4lConf(nodeProfile.Ptp4lConf)
+
+	if nodeProfile.Interface != nil && *nodeProfile.Interface != "" {
+		ifaceSection := fmt.Sprintf("[%s]\n", *nodeProfile.Interface)
+		output.sections[ifaceSection] = *new(ptp4lconfSection)
+	}
+
+
+	uds_address := fmt.Sprintf("uds_address %s\n", socketPath)
+	message_tag := fmt.Sprintf("message_tag [%s]\n", output.ifaceString)
+	section := output.sections["[global]"]
+	section.options = append(section.options, message_tag)
+	section.options = append(section.options, uds_address)
+	output.sections["[global]"] = section
+
+
+
+	*nodeProfile.Ptp4lConf = renderPtp4lConf(nodeProfile.Ptp4lConf, &output)
```

Comment:
this needs to be `err := output.renderPtp4lConf(nodeProfile.Ptp4lConf)` then you can just use 

output.conf  and output.ifaceString